workshops/payments/views.py

```diff
@@ -303,13 +305,31 @@ def post(self, request, *args, **kwargs):
                 body += '<li>%s (%s): %s</li>' % (oi.name, oi.email, oi.rate)
             body += '</ul></div>'
             body += '</html>'
-
+            rates = {}
+            for ticket in all_orders:
+                rate = ticket.rate
+                if rate not in rates:
+                    rates[rate] = 1
+                else:
+                    rates[rate] += 1
+            for rate in rates:
+                rates[rate] = [rates[rate], rate.price * rates[rate]]
+            params = {
+                'today': timezone.now(),
+                'workshop': workshop,
+                'order': order,
+                'rates': rates,
+                'total': '$' + str(order.order_total),
+                'request': request,
+            }
+            invoice = render('../templates/payments/invoice.html', params)
             msg = EmailMultiAlternatives(subject, plaintext_msg,
                                          'noreply@qiime2.org',
                                          to=[order.contact_email],
                                          cc=[i.email for i in all_orders],
                                          bcc=[i[1] for i in settings.ADMINS])
             msg.attach_alternative(body, "text/html")
+            msg.attach('invoice.pdf', invoice, 'application/pdf')
```

Comment:
Can you also add a new URL endpoint for "testing" the HTML template behind this PDF. I will leave the design and "how" as an exercise for you, but, if you get stuck or need clarification please find me.

Reply:
By "testing" it do you mean loading the html in the browser and looking at it before it's converted into a pdf?

Reply:
Oh this is beautiful

![image](https://user-images.githubusercontent.com/10642616/63189924-f3e38380-c019-11e9-982c-d25691af191b.png)

Reply:
That's just displaying in the '/submit/' view right now (the submit view just displayed a placeholder before), by your comment, it seems like you want it in its own new view correct?